code/controllers/subsystem/gravity.dm

```diff
@@ -0,0 +1,180 @@
+var/datum/subsystem/gravity/SSgravity
+var/global/legacy_gravity = FALSE
+var/global/mob_base_gravity_slip_chance = 10
+var/global/mob_base_gravity_fall_chance = 5
+var/global/mob_handhold_gravity_slip_chance = 6
+var/global/mob_handhold_gravity_fall_chance = 3
+var/global/mob_gravity_strength_slip_mod = 3
+var/global/mob_gravity_strength_fall_mod = 1.5
+
+
+/datum/subsystem/gravity
+	name = "Gravity"
+	priority = 75
+	wait = 1
+	init_order = -100
+	flags = SS_KEEP_TIMING | SS_BACKGROUND
+
+	var/list/currentrun = list()
+	var/list/currentrun_manual = list()
+	var/recalculation_cost = 0
+	var/do_purge = FALSE
+	var/purge_interval = 600
+	var/purge_tick = 0
+	var/purging = FALSE
+	var/error_no_atom = 0
+	var/error_mismatched_area = 0
+	var/error_mismatched_turf = 0
+	var/error_no_area = 0
+	var/error_no_turf = 0
+	var/list/purging_atoms = list()
+	var/list/area_blacklist_typecache
+	var/list/area_blacklist = list(/area/lavaland, /area/mine, /area/centcom)
+	var/mob_slip_chance = 0
+	var/mob_fall_chance = 0
+	var/mob_slip_chance_handhold = 0
+	var/mob_fall_chance_handhold = 0
+	var/mob_slip_chance_mod = 3
+	var/mob_fall_chance_mod = 1.5
+	var/normal_gravity = 0
+
+/datum/subsystem/gravity/New()
+	NEW_SS_GLOBAL(SSgravity)
+
+/datum/subsystem/gravity/Initialize()
+	area_blacklist_typecache = typecacheof(area_blacklist)
+	mob_slip_chance = mob_base_gravity_slip_chance
+	mob_fall_chance = mob_base_gravity_fall_chance
+	mob_slip_chance_handhold = mob_handhold_gravity_slip_chance
+	mob_fall_chance_handhold = mob_handhold_gravity_fall_chance
+	mob_slip_chance_mod = mob_gravity_strength_slip_mod
+	mob_fall_chance_mod = mob_gravity_strength_fall_mod
+	normal_gravity = legacy_gravity
+	. = ..()
+
+/datum/subsystem/gravity/Recover()
+	Initialize()	//Force init..
```

Comment:
This is going to reprint the initialization message. move the assignments into a separate proc if you have to do it like this